### PR TITLE
TF-886 [Security] [Web] Password leaks

### PR DESCRIPTION
**Pull request by dab246**
#886 

### Resolved

https://user-images.githubusercontent.com/80730648/188792549-02607d58-0feb-4696-b8b5-4d03e960dc38.mov

https://user-images.githubusercontent.com/80730648/188701766-9192d99b-add1-46f4-97b5-3cd74922d9fc.mov

### CODE REVIEW

**Comment by chibenwa** [comment]
Please see if you can make @tuanlc try this changeset and validate that it is ok. This sounds like a good thing to me that both of you meet to see this during "super tuesday".

I also want to make sure that Hive is also used to store OIDC acces / refresh tokens...

File: lib/features/caching/config/hive_cache_config.dart [hunk 1]
```diff
@@ -31,6 +37,24 @@ class HiveCacheConfig {
     }
   }
 
+  static Future<Uint8List?> getEncryptionKey() async {
+    final sharedPreference = Get.find<SharedPreferences>();
+    final containsEncryptionKey = sharedPreference.containsKey(LoginConstant.keyHiveEncrypt);
+    if (!containsEncryptionKey) {
+      final key = Hive.generateSecureKey();
+      await sharedPreference.setString(LoginConstant.keyHiveEncrypt, base64UrlEncode(key));
```
Comment (hoangdat) [inline lib/features/caching/config/hive_cache_config.dart:45]: why not store `encryptionKey` in Hive also ?

File: lib/features/caching/authentication_info_cache_client.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    });
+  }
+
+  @override
+  Future<Box<AuthenticationInfoCache>> openBox() async {
+    return Future.sync(() async {
+      final encryptionKey = await HiveCacheConfig.getEncryptionKey();
+      return Hive.openBox<AuthenticationInfoCache>(
+          tableName,
+          encryptionCipher: HiveAesCipher(encryptionKey!));
```
Comment (hoangdat) [inline lib/features/caching/authentication_info_cache_client.dart:107]: ❤️

Comment (hoangdat) [reply lib/features/caching/authentication_info_cache_client.dart:107]: please also add `encrypted box` for token/OIDC also

File: lib/features/caching/config/hive_cache_config.dart (hunk 1 above)
Comment (hoangdat) [reply lib/features/caching/config/hive_cache_config.dart:45]: or use [flutter_secure_storage](https://pub.dev/packages/flutter_secure_storage)

**Comment by dab246** [comment]
> OIDC acces / refresh tokens.

Oke. I understand. About `OIDC access/refresh token` we have already saved it to the `hive` database before. You don't need to worry about that.

File: lib/features/caching/config/hive_cache_config.dart (hunk 1 above)
Comment (dab246) [reply lib/features/caching/config/hive_cache_config.dart:45]: > or use [flutter_secure_storage](https://pub.dev/packages/flutter_secure_storage)

`flutter_secure_storage` requires some configuration for platforms like. The `Android` side is `minSdkVersion is 18`. I don't want to change it so I don't use it at this time.

Comment (dab246) [reply lib/features/caching/config/hive_cache_config.dart:45]: > why not store `encryptionKey` in Hive also ?

I will look into whether hive should be used.

**Comment by dab246** [comment]
> Please see if you can make @tuanlc try this changeset and validate that it is ok. This sounds like a good thing to me that both of you meet to see this during "super tuesday".
> 
> I also want to make sure that Hive is also used to store OIDC acces / refresh tokens...

Oke. I understand. About OIDC access/refresh token we have already saved it to the hive database before. You don't need to worry about that.

**Review by hoangdat — APPROVED** [review]

**Review by ManhNTX — APPROVED** [review]

**Comment by hoangdat** [comment]
please let @tuanlc try this version at the beginning of next week.

**Pull request merged**